tests/map/test-map.c

```diff
@@ -0,0 +1,102 @@
+#include <assert.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+#include "map.h"
+
+static void swap(int *x, int *y)
+{
+    int tmp = *x;
+    *x = *y;
+    *y = tmp;
+}
+
+enum { N_NODES = 10000 };
+
+/* return 0 on success; non-zero values on failure */
+static int test_map_mixed_operations()
+{
+    map_t tree = map_init(int, int, map_cmp_uint);
+
+    int key[N_NODES], val[N_NODES];
+
+    /*
+     *  Generate data for insertion
+     */
+    for (int i = 0; i < N_NODES; i++) {
+        key[i] = i;
+        val[i] = i + 1;
+    }
+
+    /* TODO: This is not a reconmended way to randomize stuff, just a simple
+     * test. Using MT19937 might be better
+     */
+    for (int i = 0; i < N_NODES; i++) {
+        int pos_a = rand() % N_NODES;
+        int pos_b = rand() % N_NODES;
+        swap(&key[pos_a], &key[pos_b]);
+        swap(&val[pos_a], &val[pos_b]);
+    }
+
+    /* add first 1/2 items */
+    for (int i = 0; i < N_NODES / 2; i++) {
+        map_iter_t my_it;
+        map_insert(tree, key + i, val + i);
+        map_find(tree, &my_it, key + i);
+        if (!my_it.node)
+            return 1; /* test failed */
```

Comment:
Don't return directly. Instead, you shall release the allocated resources. Using `goto` may be reasonably acceptable. See https://hackmd.io/@sysprog/c-control-flow